### PR TITLE
Light client 

### PR DESCRIPTION
**Pull request by alexvandesande**
Second PR, focusing on light client issues only. 

Review but do not merge before more group discussion.

### CODE REVIEW

**Comment by alexvandesande** [comment]
goes better after https://github.com/ethereum/mist/pull/1545/files

File: modules/ethereumNode.js [hunk 1]
```diff
@@ -130,7 +139,7 @@ class EthereumNode extends EventEmitter {
                 log.info(`Network: ${this.defaultNetwork}`);
```
Comment (frozeman) [inline modules/ethereumNode.js]: Would be good to add: `  log.info(`Mode: ${this.defaultMode}`);`

File: modules/ethereumNode.js [hunk 2]
```diff
@@ -140,20 +149,20 @@ class EthereumNode extends EventEmitter {
...
-            log.info('Restart node', newType, newNetwork);
+            log.info('Restart node..', newType, newNetwork, newMode);
 
             return this.stop()
                 .then(() => {
                     Windows.loading.show();
                 })
                 .then(() => {
-                    return this._start(newType || this.type, newNetwork || this.network);
+                    return this._start(newType || this.type, newNetwork || this.network, newMode || this.nodeMode);
```
Comment (frozeman) [inline modules/ethereumNode.js]: Whats this `this.nodeMode` ? i never see this variable created,shouldnt it be `this.mode` or something?

**Comment by alexvandesande** [comment]
@frozeman addressed both issues

File: modules/ethereumNode.js [hunk 3]
```diff
@@ -162,7 +163,7 @@ class EthereumNode extends EventEmitter {
                     Windows.loading.show();
                 })
                 .then(() => {
-                    return this._start(newType || this.type, newNetwork || this.network, newMode || this.nodeMode);
+                    return this._start(newType || this.type(), newNetwork || this.network(), newMode || this.mode());
```
Comment (frozeman) [inline modules/ethereumNode.js]: thats not functions, this are get variables.
So just `this.mode`

Comment (alexvandesande) [inline modules/ethereumNode.js]: I've tested and `this.mode()` was returning a value and `this.mode` wasn't. Are you sure this isn't being made into a function by https://github.com/ethereum/mist/pull/1547/files#diff-409860c680be3333b2089d577f193cf5R65 ?

Will test more this part

**Comment by evertonfraga** [comment]
- When the light client is far behind and I'm on the app screen (not splash), mist hits high CPU usage until the node catches up. That may cause unresponsiveness on slower machines (mine is a quad-core 2.2 GHz)
- Mist slowness was on the main thread, making the actual interface sluggish
- When I'm on light client mode and disconnect from internet and connect again, it didn't seem to automatically reconnect – or there weren't many peers available on my tests.

**Comment by alexvandesande** [comment]
superseded by https://github.com/ethereum/mist/pull/3052

**Pull request closed without merging**